modules/core/arrow-annotations-processor/build.gradle

```diff
@@ -5,6 +5,7 @@ apply plugin: 'kotlin-kapt'
 
 dependencies {
     compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlinVersion"
+    compile "com.squareup:kotlinpoet:$kotlinPoetVersion"
```

Comment:
`compileOnly`?

Reply:
I tried it and it didn't generate the files, I'm not sure, I could have something setup incorrectly too because I have to manually add the generated dirs as source dirs still

Reply:
it's fine to be `compile` because this is the annotation processor and itself it's compile only through `kapt` to it's dependent projects